astropy/config/paths.py

```diff
@@ -185,6 +163,30 @@ def wrapper(*args: P.args, **kwargs: P.kwargs) -> None:
 
         return wrapper
 
+    @classmethod
+    def get_dir_path(cls, rootname: str) -> Path:
+        if (xch := cls._temp_path) is not None:
+            path = xch / rootname
+            if not path.is_file():
+                path.mkdir(exist_ok=True)
+            return path.resolve()
+
+        if (
+            (xdg_dir := os.getenv(f"XDG_{cls._fallback_dirname.upper()}_HOME"))
+            is not None
+            and (xch := Path(xdg_dir)).exists()
+            and not (xchpth := xch / rootname).is_symlink()
+        ):
+            if xchpth.exists():
+                return xchpth.resolve()
+
+            # symlink will be set to this if the directory is created
+            linkto = xchpth
+        else:
+            linkto = None
+
+        return _find_or_create_root_dir(cls._fallback_dirname, linkto, rootname)
```

Comment:
If `_find_or_create_root_dir` were a class method too then we wouldn't have to pass `cls._fallback_dirname` as an argument when we call it.